File: python/53. Maximum Subarray.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        maxsofar = maxsum = nums[0]
        for val in nums[1:]:
            maxsofar = max(maxsofar + val, val)
            maxsum = max(maxsum, maxsofar)
        return maxsum

File: cpp/51-100/53. Maximum Subarray.cpp
class Solution {
public:
	int maxSubArray(vector<int>& nums) {
		int res = 0;
		if (nums.empty())
			return res;
		res = nums[0];
		int maxsum = res;
		int n = nums.size();
		for (int i = 1; i < n; i++) {

			if (res < 0)
				res = nums[i];
			else
			{

				res += nums[i];
			}
			maxsum = max(res, maxsum);
		}
		return maxsum;
	}
};